src/core/dataset.rs

```diff
@@ -151,24 +154,34 @@ pub trait Dataset<T: Number, U: Number>: std::fmt::Debug + Send + Sync {
             .unwrap()
             .0]
     }
+
+    // TODO: Clean up the names on these
+    fn set_reordered_indices(&mut self, indices: &[usize]);
+    // This method will fail if an `reorder` has not been called.
+    fn get_reordered_index(&self, i: usize) -> usize;
```

Comment:
This type should be Result<usize, E> for some error type E (which could be String, or Box<Error>)